Test/git-issues/git-issue-977.dfy

```diff
@@ -181,6 +181,31 @@ lemma Test7(k: nat, n: int)
 // ----- Inline/trigger issues for prefix predicates --------------------------------------------------------
 
 module PrefixBodyInlining {
+  /* To generate more specific error messages, every user-defined assertion is processed by
+   * TrSplitExpr. One of the things TrSplitExpr does is to inline predicates. It is not good
+   * to inline a predicate call P(e) if the body of P uses P's argument in a way that makes it
+   * considered to be a trigger (aka matching pattern). For example, consider predicate
+   * GreatestManualOrd(k, num) defined above, whose body contains
+   *
+   *         forall m :: m < k ==> GreatestManualOrd(m, num)
+   *
+   * The (automatically generated) trigger for this quantifier is GreatestManualOrd(m, num).
+   * Note that this trigger includes GreatestManualOrd's second argument, so if that
+   * argument is something that is not allowed in a trigger, then by inlining the call,
+   * TrSplitExpr would generate malformed Boogie code.
```

Comment:
I don't understand this paragraph.  If the second argument is something that is not allowed in a trigger, then `GreatestManualOrd(m, num)` is already malformed, isn't it?